addon/components/listbox.js

```diff
@@ -0,0 +1,306 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { guidFor } from '@ember/object/internals';
+import { action } from '@ember/object';
+import { debounce } from '@ember/runloop';
+
+const ACTIVATE_NONE = 0;
+const ACTIVATE_FIRST = 1;
+const ACTIVATE_LAST = 2;
+
+export default class ListboxComponent extends Component {
+  @tracked activeOptionIndex;
+  activateBehaviour = ACTIVATE_NONE;
+  buttonElement;
+  guid = `${guidFor(this)}-headlessui-listbox`;
+  @tracked _isOpen = false;
+  labelElement;
+  optionsElement;
+  optionElements = [];
+  optionValues = {};
+  search = '';
+  @tracked selectedOptionIndex;
+
+  get activeOptionGuid() {
+    return this.optionElements[this.activeOptionIndex]?.id;
+  }
+
+  get isDisabled() {
+    return !!this.args.disabled;
+  }
+
+  get selectedOptionGuid() {
+    return this.optionElements[this.selectedOptionIndex]?.id;
+  }
+
+  get isOpen() {
+    return this._isOpen;
+  }
+
+  set isOpen(isOpen) {
+    if (isOpen) {
+      this.activeOptionIndex = undefined;
+      this.selectedOptionIndex = undefined;
+      this.optionElements = [];
+      this.optionValues = {};
+      this._isOpen = true;
+    } else {
+      this._isOpen = false;
+    }
+  }
+
+  @action
+  closeListbox() {
+    this.isOpen = false;
+  }
+
+  @action
+  handleButtonClick(e) {
+    if (e.button !== 0) return;
+    this.activateBehaviour = ACTIVATE_NONE;
+    this.isOpen = !this.isOpen;
+  }
+
+  @action
+  handleClickOutside(e) {
+    if (e.button !== 0) return;
```

Comment:
What does this check do?

Reply:
Mostly to make a right-click test pass. It doesn't apply in real world 😆 

It has been removed anyway as part of https://github.com/GavinJoyce/ember-headlessui/pull/76/commits/660f5cf28fdee9d2c16ba23f2ba3a2a4205ed3ff